src/test/java/com/commercetools/sync/sdk2/taxcategories/TaxCategorySyncTest.java

```diff
@@ -0,0 +1,480 @@
+            .name(name)
+            .rates(
+                TaxRateDraftBuilder.of()
+                    .name(name)
+                    .amount(2.0)
+                    .includedInPrice(false)
+                    .country(CountryCode.FR.getAlpha2())
+                    .state("LYON")
+                    .build(),
+                TaxRateDraftBuilder.of()
+                    .name(name)
+                    .amount(2.0)
+                    .includedInPrice(false)
+                    .country(CountryCode.FR.getAlpha2())
+                    .state("PARIS")
+                    .build(),
+                taxRateGermanyBerlin,
+                taxRateGermanyBerlin)
+            .description("desc")
+            .key("someKey")
+            .build();
+
+    final AtomicReference<String> callback = new AtomicReference<>(null);
+    final TaxCategorySyncOptions syncOptions =
+        TaxCategorySyncOptionsBuilder.of(mock(ProjectApiRoot.class))
+            .errorCallback(
+                (exception, taxDraft, entry, actions) -> callback.set(exception.getMessage()))
+            .build();
+    final TaxCategorySync sync = new TaxCategorySync(syncOptions, taxCategoryService);
+    final TaxCategory taxCategory = mock(TaxCategory.class);
+
+    when(taxCategory.getId()).thenReturn("id");
+    when(taxCategory.getKey()).thenReturn("someKey");
+
+    when(taxCategoryService.fetchMatchingTaxCategoriesByKeys(any()))
+        .thenReturn(completedFuture(new HashSet<>(singletonList(taxCategory))));
+    when(taxCategoryService.updateTaxCategory(any(), any()))
+        .thenReturn(completedFuture(taxCategory));
+
+    final TaxCategorySyncStatistics result =
+        sync.sync(singletonList(draft)).toCompletableFuture().join();
+
+    assertAll(
+        () -> assertThat(result.getProcessed().get()).isEqualTo(1),
+        () -> assertThat(result.getUpdated().get()).isEqualTo(0),
+        () -> assertThat(result.getFailed().get()).isEqualTo(1),
+        () ->
+            assertThat(callback.get())
+                .contains(
+                    format(
+                        "Tax rate drafts have duplicated country codes and states. Duplicated "
+                            + "tax rate country code: '%s'. state : '%s'. Tax rate country codes and states are "
+                            + "expected to be unique inside their tax category.",
+                        CountryCode.DE, "BERLIN")));
+  }
+
+  @Test
+  void sync_WithDuplicatedCountryCode_ShouldNotBuildActionAndTriggerErrorCallback() {
+    final String name = "DuplicatedName";
+    final TaxRateDraft taxRateFrance =
+        TaxRateDraftBuilder.of()
+            .name(name)
+            .amount(2.0)
+            .includedInPrice(false)
+            .country(CountryCode.FR.getAlpha2())
+            .build();
+    final TaxCategoryDraft draft =
+        TaxCategoryDraftBuilder.of()
+            .name(name)
+            .rates(taxRateFrance, taxRateFrance)
```

Comment:
Isn't this case similar to the previous one `sync_WithDuplicatedState_ShouldNotBuildActionAndTriggerErrorCallback`?

Reply:
Yes, it is. It's just checkin different error message... this is already checked in batchvalidatortest, so i can remove from here.